daml-lf/validation/src/test/scala/com/digitalasset/daml/lf/validation/TypingSpec.scala

```diff
@@ -302,156 +308,274 @@ class TypingSpec extends WordSpec with TableDrivenPropertyChecks with Matchers {
+          { case _: ETypeMismatch => },
         // ScnCommit
-        E"Λ (τ : ⋆ → ⋆). λ (e₁: Party) (e₂: Update τ) → (( commit @τ e₁ e₂ ))",
-        E"Λ (τ : ⋆) (σ : ⋆). λ (e₁: σ) (e₂: Update τ) → (( commit @τ e₁ e₂ ))",
-        E"Λ (τ : ⋆) (σ : ⋆). λ (e₁: Party) (e₂: Update σ) → (( commit @τ e₁ e₂ ))",
-        E"Λ (τ : ⋆) (σ : ⋆). λ (e₁: Party) (e₂: σ) → (( commit @τ e₁ e₂ ))",
-        E"Λ (τ : ⋆) (σ : ⋆). λ (e₁: Party) (e₂: Update τ) → (( commit @σ e₁ e₂ ))",
+        E"Λ (τ : ⋆ → ⋆). λ (e₁: Party) → ⸨ commit @τ e₁ nothing ⸩" -> //
+          { case _: EKindMismatch => },
+        E"Λ (τ : ⋆) (σ : ⋆). λ (e₁: σ) (e₂: Update τ) → ⸨ commit @τ e₁ e₂ ⸩" -> //
+          { case _: ETypeMismatch => },
+        E"Λ (τ : ⋆) (σ : ⋆). λ (e₁: Party) (e₂: Update σ) → ⸨ commit @τ e₁ e₂ ⸩" -> //
+          { case _: ETypeMismatch => },
+        E"Λ (τ : ⋆) (σ : ⋆). λ (e₁: Party) (e₂: σ) → ⸨ commit @τ e₁ e₂ ⸩" -> //
+          { case _: ETypeMismatch => },
+        E"Λ (τ : ⋆) (σ : ⋆). λ (e₁: Party) (e₂: Update τ) → ⸨ commit @σ e₁ e₂ ⸩" -> //
+          { case _: ETypeMismatch => },
         // ScnMustFail
-        E"Λ (τ : ⋆ → ⋆). λ (e₁: Party) (e₂: Update τ) → (( must_fail_at @τ e₁ e₂ ))",
-        E"Λ (τ : ⋆) (σ : ⋆). λ (e₁: σ) (e₂: Update τ) → (( must_fail_at @τ e₁ e₂ ))",
-        E"Λ (τ : ⋆) (σ : ⋆). λ (e₁: Party) (e₂: Update σ) → (( must_fail_at @τ e₁ e₂ ))",
-        E"Λ (τ : ⋆) (σ : ⋆). λ (e₁: Party) (e₂: σ) → (( must_fail_at @τ e₁ e₂ ))",
-        E"Λ (τ : ⋆) (σ : ⋆). λ (e₁: Party) (e₂: Update τ) → (( must_fail_at @σ e₁ e₂ ))",
+        E"Λ (τ : ⋆ → ⋆). λ (e₁: Party) → ⸨ must_fail_at @τ e₁ nothing ⸩" -> //
+          { case _: EKindMismatch => },
+        E"Λ (τ : ⋆) (σ : ⋆). λ (e₁: σ) (e₂: Update τ) → ⸨ must_fail_at @τ e₁ e₂ ⸩" -> //
+          { case _: ETypeMismatch => },
+        E"Λ (τ : ⋆) (σ : ⋆). λ (e₁: Party) (e₂: Update σ) → ⸨ must_fail_at @τ e₁ e₂ ⸩" -> //
+          { case _: ETypeMismatch => },
+        E"Λ (τ : ⋆) (σ : ⋆). λ (e₁: Party) (e₂: σ) → ⸨ must_fail_at @τ e₁ e₂ ⸩" -> //
+          { case _: ETypeMismatch => },
+        E"Λ (τ : ⋆) (σ : ⋆). λ (e₁: Party) (e₂: Update τ) → ⸨ must_fail_at @σ e₁ e₂ ⸩" -> //
+          { case _: ETypeMismatch => },
         // ScnPass
-        E"Λ (σ : ⋆). λ (e: σ) → (( pass e ))",
+        E"Λ (σ : ⋆). λ (e: σ) → ⸨ pass e ⸩" -> //
+          { case _: ETypeMismatch => },
         // ScnGetParty
-        E"Λ (σ : ⋆). λ (e: σ) → (( sget_party e ))",
+        E"Λ (σ : ⋆). λ (e: σ) → ⸨ sget_party e ⸩" -> //
+          { case _: ETypeMismatch => },
         // ScnEmbedExpr
-        E"Λ (τ : ⋆) (σ : ⋆). λ (e : Scenario σ) → (( sembed_expr @τ e ))",
-        E"Λ (τ : ⋆) (σ : ⋆). λ (e : σ) → (( sembed_expr @τ e ))",
+        E"Λ (τ : ⋆) (σ : ⋆). λ (e : Scenario σ) → ⸨ sembed_expr @τ e ⸩" -> //
+          { case _: ETypeMismatch => },
+        E"Λ (τ : ⋆) (σ : ⋆). λ (e : σ) → ⸨ sembed_expr @τ e ⸩" -> //
+          { case _: ETypeMismatch => },
```

Comment:
I don't really see the point of this one. In fact I think we should drop also 
```scala 
  E"Λ (τ : ⋆) (σ : ⋆). λ (e : Scenario σ) → ⸨ sembed_expr @τ e ⸩" -> //
          { case _: ETypeMismatch => },
```

Reply:
I have experience working in compiler codebases where some small errors like this are not caught -- i.e. the distinction between Scenario and Update is not caught, because of an oversight. 😢 

This is similar to the test cases above where I added a `Optional` instead of `List`, for example.